### PR TITLE
ecdsa: add `SignatureWithOid`

### PR DESCRIPTION
**Pull request by tarcieri**
Adds a signature type which is paramaterized by an RFC5758-defined OID identifying the variant of ECDSA used to produce the signature.

cc @baloo @lumag

### CODE REVIEW

**Comment by tarcieri** [comment]
The real benefit of this will be dynamically dispatching the hash function to use for verification, but I've left that for a followup PR.

File: ecdsa/src/lib.rs [hunk 1]
```diff
@@ -500,3 +500,194 @@ where
...
+        AlgorithmIdentifierOwned {
+            oid: self.oid,
+            parameters: None,
+        }
+    }
+}
+
+/// Get the ECDSA OID for a given digest OID.
+#[cfg(feature = "digest")]
+const fn ecdsa_oid_for_digest(digest_oid: ObjectIdentifier) -> Option<ObjectIdentifier> {
```
Comment (baloo) [inline ecdsa/src/lib.rs:685]: :heart:

Comment (tarcieri) [reply ecdsa/src/lib.rs:685]: It's private, but accessible via constants (with compile-time failures if given a non-SHA-2 OID, at that)

Comment (baloo) [reply ecdsa/src/lib.rs:685]: I was looking how to do that over in RSA, but this is very cool!

**Pull request merged**